metadata/src/utils/retain.rs

```diff
@@ -301,44 +383,34 @@ mod tests {
 
     #[test]
     fn retain_one_pallet() {
-        let metadata_cache = load_metadata();
+        let metadata_cache = load_metadata_polkadot();
 
         // Retain one pallet at a time ensuring the test does not panic.
         for pallet in metadata_cache.pallets() {
+            let original_meta = metadata_cache.clone();
             let mut metadata = metadata_cache.clone();
-
             retain_metadata(
                 &mut metadata,
                 |pallet_name| pallet_name == pallet.name(),
                 |_| true,
             );
-
             assert_eq!(metadata.pallets.len(), 1);
             assert_eq!(
                 &*metadata.pallets.get_by_index(0).unwrap().name,
                 pallet.name()
             );
 
-            let id = metadata.outer_enums().call_enum_ty;
-            let ty = metadata.types.resolve(id).unwrap();
-            let num_variants = if pallet.call_ty_id().is_some() { 1 } else { 0 };
-            assert_matches!(&ty.type_def, TypeDef::Variant(variant) if variant.variants.len() == num_variants);
-
-            let id = metadata.outer_enums().error_enum_ty;
-            let ty = metadata.types.resolve(id).unwrap();
-            let num_variants = if pallet.error_ty_id().is_some() { 1 } else { 0 };
-            assert_matches!(&ty.type_def, TypeDef::Variant(variant) if variant.variants.len() == num_variants);
-
-            let id = metadata.outer_enums().event_enum_ty;
-            let ty = metadata.types.resolve(id).unwrap();
-            let num_variants = if pallet.event_ty_id().is_some() { 1 } else { 0 };
-            assert_matches!(&ty.type_def, TypeDef::Variant(variant) if variant.variants.len() == num_variants);
+            assert!(
+                metadata.types.types.len() < original_meta.types.types.len(),
+                "Stripped metadata must have less retained types than the non-stripped one: stripped amount {}, original amount {}",
+                metadata.types.types.len(), original_meta.types.types.len()
+            );
         }
     }
 
     #[test]
     fn retain_one_runtime_api() {
-        let metadata_cache = load_metadata();
+        let metadata_cache = load_metadata_polkadot();
```

Comment:
Nit: I'd call this just `metadata` too (not really this PR but hey :))

Reply:
renamed `load_metadata_polkadot` to `load_metadata` and introduced `load_metadata_custom` to load different files